### PR TITLE
ContributionReward enhancements

### PR DESCRIPTION
**Pull request by dkent600**
Upgraded to Arc v34.

Cleaned up jslint packaging and config

Added functions to ContributionReward:

- redeemExternalToken
- redeemReputation
- redeemNativeToken
- redeemEther
- getDaoProposals
- getBeneficiaryRewards

### CODE REVIEW

File: lib/contracts/contributionreward.ts [hunk 1]
```diff
@@ -157,6 +157,213 @@ export class ContributionRewardWrapper extends ExtendTruffleContract {
...
+    const tx = await this.contract.redeemExternalToken(
+      options.proposalId,
+      options.avatar,
+    );
+
+    return new ArcTransactionResult(tx);
+  }
+
+  /**
+   * Redeem reputation reward for the beneficiary of the proposal
```
Comment (tibetsprague) [inline lib/contracts/contributionreward.ts]: oh interesting, so anyone can do the redemption and decide to pay the gas for it?

File: lib/contracts/contributionreward.ts [hunk 2]
```diff
@@ -157,6 +157,213 @@ export class ContributionRewardWrapper extends ExtendTruffleContract {
...
+
+    return proposals;
+  }
+
+  /**
+   * Return a list of executed (passed) proposals that have rewards
+   * waiting to be redeemed by the given beneficiary.
+   * @param opts
+   */
+  public async getBeneficiaryRewards(opts = {}): Promise<Array<ProposalRewards>> {
```
Comment (tibetsprague) [inline lib/contracts/contributionreward.ts]: maybe should say "unredeemed" in the name?

File: lib/contracts/contributionreward.ts (hunk 1 above)
Comment (dkent600) [reply lib/contracts/contributionreward.ts]: Apparently.  I remarked (to myself) about that too.  Same for the other reward types.

File: lib/contracts/contributionreward.ts (hunk 2 above)
Comment (dkent600) [reply lib/contracts/contributionreward.ts]: OK, I took the liberty of keeping the name but having it also return the total amounts redeemable.

**Pull request merged**